frontend/src/metabase/dashboard/actions/tabs.ts

```diff
@@ -293,9 +293,14 @@ export const tabsReducer = createReducer<DashboardState>(
         }
 
         // 1. Replace temporary with real dashcard ids
-        const prevCards = prevDash.ordered_cards.filter(
-          id => !state.dashcards[id].isRemoved,
-        );
+        const prevCards = prevDash.ordered_cards.filter(id => {
+          const dashcard = state.dashcards[id];
+          const isNotRemoved = !dashcard.isRemoved;
+          const isQuestionCard = dashcard.card_id != null;
+
+          return isNotRemoved && isQuestionCard;
+        });
+
         prevCards.forEach((oldId, index) => {
           state.dashcardData[newCards[index].id] = state.dashcardData[oldId];
```

Comment:
:x:

I'm pretty sure that source of the problems is here: there is an assumption that `index` will point to the same thing in `prevCards` and `newCards` arrays.

I followed reproduction steps from PR description, and then did this:

1. Refreshed the page
2. Edit dashboard
3. Add new chart to the dashboard
4. Drag & drop the chart to be the first chart in the dashboard
5. Save it

The new chart stays in loading state forever. This bug is not present in `master`.

https://github.com/metabase/metabase/assets/6830683/26a4f942-83ff-4b02-b8d3-61888e8150af

Reply:
nice catch, I was afraid of breaking something 100%, please check cd6fe55fef733f144a19a496ac28c3068a5b1613

Reply:
This error about reading "id" is not from here, but from dashboard reducers. Try the same steps to get an error on master branch but in Firefox - you'll see it